### DISCUSSION TITLE
[keycloak] Supabase signout does not signout at provider

### DISCUSSION THREAD

**Discussion by olee** [supabase/supabase#16316] [Questions]
I have an application which automatically signs in users with a connected keycloak login provider.
However, when I call `supabase.auth.signOut()` in my nextjs app, it only removes credentials from the nextjs app and not from keycloak.
This causes the user to be logged in directly again after the page reloaded.

One solution I managed to find is redirecting to `http://<keycloak>/realms/<realm>/protocol/openid-connect/logout?client_id=<client>&post_logout_redirect_uri=http://localhost:3000`.
However, this will then require the user to confirm the logout once more by clicking a logout button on keycloak side.

To skip this confirmation, I would have to add a `id_token_hint=<id_token>` to the request, but afaik I do not have access to this one as it is stored on supabase side.

Any idea?

**Comment by hf** [comment]
Hey, the ID token is not stored at all on the Supabase side. In general, sign-in through OAuth with Supabase Auth (and many other OAuth-enabled Auth servers) do not really handle the logout scenario. At best what is handled is logout at the identity provider side (Keycloak in your use case) which sometimes gets propagated to the service provider side (Supabase Auth). We also don't support this use case as well.
